### DISCUSSION TITLE
Lightweight version of some commands?

### DISCUSSION THREAD

**Discussion by Lorthiz** [libvips/libvips#2514] [Q&A] [answered]
Hello, 
The libvibs weight around 70MB, but I wrote a simple wrapper to help me cut image into pyramid so the only command I am using is `dzsave` by calling "bin/vips.exe dzsave {my settings}". Is it possible to somehow strip executable of additional functionalites? Or that would require downloading source code and striping it directly in the sources then recompiling it on my own?

**Comment by jcupitt** [comment]
Hi @Lorthiz,

Sure, you can build your own libvips with many optional elements removed. The core library is only about 2.5 MB.

I would take a look at the build scripts and customise them to your needs:

https://github.com/libvips/build-win64-mxe

The `web` binary is only about 50mb and includes dzsave:

https://github.com/libvips/build-win64-mxe/releases/download/v8.11.4/vips-dev-w64-web-8.11.4.zip

Perhaps that would be small enough? Or do you need openslide loading?

**Comment by jcupitt** [comment, marked as answer]
The zips include all the headers and link libraries you need to build your own programs on top of libvips. 

If you are only running `vips.exe`, then you can remove everything except the `bin` directory. I see:

```
john@banana ~/vips-dev-8.11 $ du -sk .
51636	.
john@banana ~/vips-dev-8.11 $ du -sk bin
25380	bin
```

So the whole thing is 50 MB, `bin` is 25 MB.

**Comment by Lorthiz** [comment]
Thanks!
In the version I am using it was 20mb after removing other folders. And after packing it with 7z with my wrapper now it weights only around 6,5mb compared to 13mb before(and even 50mb earlier when I added one map by mistake to bin folder :P). And that is something  
 what I wanted to achieve - pretty lightweight tool that can be easilly passed betwen me and a few coulegues to quickly cut images. Thanks for a great lib and help! :)
